tests/serialized_test_data_generation/Makefile

```diff
@@ -1,69 +1,123 @@
+# version number or tag (determines image tag and destination directory in cloud storage)
+# -convention: <tag>-<some large conceptual version change>.<serialization statement change>.<hotfix>
+# -version number should be increased when serialization data is expected to change
+# -omit tag for "operational" serialization data, use tag for experimental serialization data
+FORTRAN_VERSION ?= jenkins-7.1.1
 
+# docker container image setup (used for running model, see ../../README.md)
+GCR_URL = us.gcr.io/vcm-ml
+COMPILED_IMAGE = $(GCR_URL)/fv3gfs-compiled:$(FORTRAN_VERSION)-serialize-gt4pydev
+RUN_DIR_CONTAINER = /rundir
+DATA_DIR_CONTAINER = $(RUN_DIR_CONTAINER)/test_data
 
+# local host setup
+CWD=$(shell pwd)
+FORTRAN_DIR = $(CWD)/../../
+CONFIGURATION_DIR = $(CWD)/configs
+RUN_DIR_HOST = $(CWD)/rundir/$(CONFIGURATION)
+DATA_DIR_HOST = $(CWD)/data/$(CONFIGURATION)
+TARFILE_HOST = $(DATA_DIR_HOST)/dat_files.tar.gz
+TARFILE_INDEX_HOST = $(DATA_DIR_HOST)/tarfile_index.txt
+USER_ID_HOST = $(shell id -u)
+GROUP_ID_HOST = $(shell id -g)
+
+# cloud storage location for pushing generated data
+STORAGE_BUCKET = gs://vcm-fv3gfs-serialized-regression-data
+DATA_DIR_BUCKET = $(STORAGE_BUCKET)/$(FORTRAN_VERSION)/$(CONFIGURATION)
+
+help:
+	@awk 'BEGIN {FS = ":.*?## "} /^[a-zA-Z0-9_-]+:.*?## / {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}' $(MAKEFILE_LIST)
+
+build_model: ## build Docker image with compiled model code for serialization
+	cd $(FORTRAN_DIR) && \
+		$(MAKE) pull_deps && \
+		        BUILD_ARGS="$(BUILD_ARGS) --build-arg serialize=true" \
+			COMPILED_IMAGE=$(COMPILED_IMAGE) \
+			COMPILE_OPTION="GT4PY_DEV=Y" \
+			$(MAKE) build_compiled
+
+setup_rundir: ## create run directory on local host using fv3config
+	@if ! command -v write_run_directory ; then \
+		echo "Error: Please install fv3config (e.g. pip install -r ../../requirements.txt)" ; \
+		exit 1 ; \
+	fi
+	@if [ -d $(RUN_DIR_HOST) ] ; then \
+		echo "Error: Run directory $(RUN_DIR_HOST) already exists. Run make clean?" ; \
+		exit 1 ; \
+	fi
+	write_run_directory $(CONFIGURATION_DIR)/$(CONFIGURATION).yml $(RUN_DIR_HOST)
+	cp submit_job.sh $(RUN_DIR_HOST)
+	cp count_ranks.py $(RUN_DIR_HOST)
+	git rev-parse HEAD > $(RUN_DIR_HOST)/fortran_sha.txt
+
+run_model: ## run model using Docker image to generate serialized data
+	@if [ ! "`docker image ls $(COMPILED_IMAGE) | wc -l`" -eq 2 ] ; then \
+		echo "Error: Docker image $(COMPILE_IMAGE) does not exist. Run make build_model?" ; \
+		exit 1 ; \
+	fi
+	@if [ -d $(DATA_DIR_HOST) ] ; then \
+		echo "Error: Data directory $(DATA_DIR_HOST) already exists. Run make clean?" ; \
+		exit 1 ; \
+	fi
+	mkdir -p $(DATA_DIR_HOST)
+	docker run -e USER_ID_HOST=$(USER_ID_HOST) -e GROUP_ID_HOST=$(GROUP_ID_HOST) \
+		--network host --rm -v $(RUN_DIR_HOST):$(RUN_DIR_CONTAINER) \
+		-v $(DATA_DIR_HOST):$(DATA_DIR_CONTAINER) $(COMPILED_IMAGE) \
+		/rundir/submit_job.sh
+	cd $(DATA_DIR_HOST) && \
+		md5sum * > md5sums.txt
```

Comment:
This isn't used, is this to give us a future option of ensuring we haven't changed the data?

Reply:
No it is. There is a `make validate_data` target which only validates the md5sum's of the generated data as compared to the GCP bucket reference.

Reply:
Oh cool! I can't seem to find 'validate_data' in this diff besides this comment. I see the validate only button now in jenkins -- nice! Though I don't see anything here for that either? Have you not pushed the change or am I missing something hidden from the default view?

Reply:
Sorry, this was not pushed yet when I wrote the comment.